server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java

```diff
@@ -0,0 +1,139 @@
+package org.infinispan.server.core.transport;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.management.AttributeNotFoundException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.CacheException;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.distexec.DefaultExecutorService;
+import org.infinispan.distexec.DistributedCallable;
+import org.infinispan.distexec.DistributedExecutorService;
+import org.infinispan.jmx.JmxUtil;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+import io.netty.channel.group.ChannelGroup;
+
+public class NettyTransportConnectionStats {
+
+   private final EmbeddedCacheManager cacheManager;
+   private final boolean isGlobalStatsEnabled;
+   private final ChannelGroup acceptedChannels;
+   private final String threadNamePrefix;
+   private final AtomicLong totalBytesWritten = new AtomicLong();
+   private final AtomicLong totalBytesRead = new AtomicLong();
+
+   public NettyTransportConnectionStats(EmbeddedCacheManager cacheManager, ChannelGroup acceptedChannels, String threadNamePrefix) {
+      this.cacheManager = cacheManager;
+      this.isGlobalStatsEnabled = cacheManager != null ? cacheManager.getCacheManagerConfiguration().globalJmxStatistics().enabled() : false;
```

Comment:
Can be simplified to `cacheManager != null && cacheManager.getCacheManagerConfiguration().globalJmxStatistics().enabled()`

Reply:
Good point!

Reply:
Done